include/glow/Runtime/HostManager/HostManager.h

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_RUNTIME_HOSTMANAGER_H
+#define GLOW_RUNTIME_HOSTMANAGER_H
+#include "glow/Graph/Graph.h"
+#include "glow/Runtime/Executor.h"
+#include "glow/Runtime/Provisioner.h"
+#include "glow/Runtime/RuntimeTypes.h"
+
+#include <llvm/ADT/StringRef.h>
+
+#include <atomic>
+#include <map>
+#include <unordered_map>
+#include <vector>
+
+namespace glow {
+namespace runtime {
+
+/// Callback signalling the result of running a function.
+using ResultCBTy = std::function<void(
+    runtime::RunIdentifierTy, runtime::ResultCode, std::unique_ptr<Context>)>;
+
+class DeviceManager;
+class Context;
+class DeviceConfig;
+class HostManager final : std::enable_shared_from_this<HostManager> {
+  /// Count of current in-flight networks being run.
+  std::atomic<size_t> activeRequestCount_{0};
+
+  /// Count of total requests, this is used as a run ID.
+  std::atomic<size_t> totalRequestCount_{0};
+
+  /// Limit maximum count of networks run at once. Hardcoded for now this should
+  /// be a configurable value.
+  const unsigned int activeRequestLimit_ = 20;
+  /// A map from a networkID to the DAG that represents the network.
+  std::unordered_map<std::string, DAGNode> networks_;
+  /// A map of DeviceManagers by deviceID.
+  std::map<DeviceIDTy, std::unique_ptr<DeviceManager>> devices_;
+  Executor executor_;
+  Provisioner provisioner_;
+
+public:
+  /// Adds the network to the host and does the necessary setup work. This
+  /// includes partitioning, provisioning, compiling and initializing backends.
+  /// Returns the networkID of the network.
```

Comment:
Is the comment about the return code correct?  It says ResultCode not network ID. Either there is a bug or the code is not readable.